src/steps/do-query/do-get.js

```diff
@@ -63,20 +64,56 @@ export default function(requestContext, responseContext, registry) {
 
 }
 
-function parseFields(fieldsParam) {
+function parseFields(fieldsParam, registry) {
   let fields;
   if(typeof fieldsParam === "object") {
     fields = {};
     let isField = (it) => !arrayContains(["id", "type"], it);
 
     for(let type in fieldsParam) {
-      let provided = parseCommaSeparatedParam(fieldsParam[type]) || [];
+      let provided = parseCommaSeparatedParam(fieldsParam[type], type, registry) || [];
       fields[type] = provided.filter(isField);
     }
   }
   return fields;
 }
 
-function parseCommaSeparatedParam(it) {
-  return it ? it.split(",").map(decodeURIComponent) : undefined;
+function parseCommaSeparatedParam(it, type, registry) {
+  return it ? it.split(",").map(f => {
+    return transformDotSeparatedPath(f, type, registry);
+  }) : undefined;
+}
+
+function transformDotSeparatedPath(it, type, registry) {
+  let parts = it.split(".");
+  let types = [type];
+  parts.forEach((p, index) => {
+    if (index < parts.length - 1) {
+      types.push(registry.dbAdapter(types[index]).constructor.getReferencedType(types[index], p));
+    }
+  });
+
+  return parts.map((part, index) => {
+    if (shouldCamelizeType(types[index], registry)) {
+      return camelize(decodeURIComponent(part));
+    }
+    return decodeURIComponent(part);
+  }).join(".");
+}
+
+function transformFilters(it, type, registry) {
```

Comment:
Hm, good point about camelizing here—even with the "simple" strategy, we get mongo directives (probably not the right word) in there, e.g. `$and`, `$in`, etc. None of them would be affected by camelization, but... I get the point. What if we retrieved an exhaustive list of field names for the resource from the adapter, and only ever transformed those? Can that be done reliably with Mongoose, accounting for virtuals and relationships and such?